File: back/src/main/java/com/back/domain/chat/chatRoomMessage/dto/BroadcastChatMessageDto.java
package com.back.domain.chat.chatRoomMessage.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.LocalDateTime;
import java.util.UUID;

public record BroadcastChatMessageDto(
        UUID messageId,
        UUID roomId,
        String senderNickname,
        UUID senderParticipantId,
        String content,
        LocalDateTime createdAt,
        @JsonProperty("isMine") boolean isMine
) {
    public BroadcastChatMessageDto(RedisChatMessageDto source, boolean isMine) {
        this(
                source.getMessageId(),
                source.getRoomId(),
                source.getSenderNickname(),
                source.getSenderParticipantId(),
                source.getContent(),
                source.getCreatedAt(),
                isMine
        );
    }
}

File: back/src/main/kotlin/com/back/domain/chat/chatRoomMessage/dto/BroadcastChatMessageDto.kt
package com.back.domain.chat.chatRoomMessage.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime
import java.util.UUID

data class BroadcastChatMessageDto(
    val messageId: UUID,
    val roomId: UUID,
    val senderNickname: String,
    val senderParticipantId: UUID,
    val content: String?,
    val createdAt: LocalDateTime,
    @get:JsonProperty("isMine") val isMine: Boolean
) {
    constructor(source: RedisChatMessageDto, isMine: Boolean) : this(
        source.messageId,
        source.roomId,
        source.senderNickname,
        source.senderParticipantId,
        source.content,
        source.createdAt,
        isMine
    )
}
